pkg/agent/trafficfilter/traffic_filter.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trafficfilter
+
+import (
+	"net"
+	"sync"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/config"
+)
+
+type Client interface {
+	Initialize(nodeConfig *config.NodeConfig) error
+	AllowNodeToPeerCIDRTraffic(peerPodCIDR *net.IPNet, peerNodeIP net.IP) error
+	Reconcile() error
+}
+
+// NewClient constructs a Client instance for iptables operations.
+func NewClient(hostGateway string, serviceCIDR *net.IPNet, encapMode config.TrafficEncapModeType) Client {
```

Comment:
Do you think better to move this func to traffic_filter_os.go, and then for Windows, just need an empty client struct.